### ISSUE TITLE
Support empty stop times according to GTFS spec

### ISSUE THREAD

**Issue by grote** [hove-io/navitia#1362]
The GTFS specification allows stop times to be empty as long as the first and the last stop have a time associated. This is currently not possible with Navitia. Stop times defined this way are just deleted with this warning:

```
sorting stoptimes of vehicle_journeys
invalid vj 4446_dst_1, the stop times are not correcly ordered stop time 1 [76500, 76500] stop time 2 [10799, 10799] we erase the VJ
invalid vj 4446_dst_2, the stop times are not correcly ordered stop time 1 [72900, 72900] stop time 2 [7199, 7199] we erase the VJ
```

This is what the [GTFS specification](https://developers.google.com/transit/gtfs/reference#stop_times_fields) says about `stop_times.txt`:

> If this stop isn't a time point, use an empty string value for the
> _arrival_time_ and _departure_time_ fields. Stops without arrival times
> will be scheduled based on the nearest preceding timed stop. To ensure
> accurate routing, please provide arrival and departure times for all stops
> that are time points. Do not interpolate stops.
> 
> You must specify arrival and departure times for the first and last stops in a trip. 

So in order to properly implement the GTFS format, Navitia should be able to handle empty stop times and interpolate the times as best as it can. A sample python implementation is available from [google/transitfeed](https://github.com/google/transitfeed/blob/master/transitfeed/trip.py#L169).

**Comment by TeXitoi** [comment]
No, the spec that you quote say explicitly to **not** interpolate! If we support this kind of stop times, we will not interpolate.

**Comment by grote** [comment]
@TeXitoi Thanks for responding, but I think there might be a misunderstanding. The person creating the GTFS feed should not interpolate stop times when they are not known. But of course a routing engine like Navita when working with the GTFS feed needs to approximate those times to be able to do trip routing. Please also observe this sentence in the specification:

> Stops without arrival times will be scheduled based on the nearest preceding timed stop.

**Comment by pbougue** [comment]
Discussion about that (more focused on OTP): https://groups.google.com/forum/#!topic/transit-developers/dwd96EwJqIc

Navitia discussion about that: https://groups.google.com/forum/#!topic/navitia/R2WOtVKuvuo
